### ISSUE TITLE
DOC: Fix quotes position in pandas.core

### ISSUE THREAD

**Issue by datapythonista** [pandas-dev/pandas#24071]
To keep out docstrings consistent, and make them easier to read, we try to keep always the same exact format. In the case of the opening and closing quotes, the right format is:
```
def foo():
    """
    Summary of the docstring.

    Returns
    -------
    str
        Explaining what is being returned.
    """
```

But for historical reasons, we have many docstrings with different format, for example:
```
def foo():
    """Summary of the docstring.

    Returns
    -------
    str
        Explaining what is being returned."""
```

Besides being inconsistent, having docstrings with the wrong format doesn't let us start validating and enforcing automatically the wanted format. So, we need to change all the wrong cases.

For this issue, all the wrong cases of methods of `pandas.core` should be fixed. The list can be obtained with the next command (the command should not report any error when this is fixed):
```
$ ./scripts/validate_docstrings.py --prefix=pandas.core --errors=GL01,GL02
pandas.core.window.Rolling.skew: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.window.Expanding.skew: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.GroupBy.aggregate: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.GroupBy.aggregate: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.GroupBy.transform: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.GroupBy.transform: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.GroupBy.pipe: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.GroupBy.first: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.GroupBy.last: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.GroupBy.max: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.GroupBy.min: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.GroupBy.prod: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.GroupBy.sum: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.DataFrameGroupBy.corr: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.DataFrameGroupBy.corr: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.DataFrameGroupBy.count: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.DataFrameGroupBy.count: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.DataFrameGroupBy.cov: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.DataFrameGroupBy.cov: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.DataFrameGroupBy.diff: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.DataFrameGroupBy.diff: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.DataFrameGroupBy.fillna: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.DataFrameGroupBy.fillna: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.DataFrameGroupBy.hist: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.DataFrameGroupBy.hist: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.DataFrameGroupBy.idxmax: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.DataFrameGroupBy.idxmax: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.DataFrameGroupBy.idxmin: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.DataFrameGroupBy.idxmin: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.DataFrameGroupBy.mad: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.DataFrameGroupBy.mad: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.DataFrameGroupBy.quantile: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.DataFrameGroupBy.quantile: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.DataFrameGroupBy.skew: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.DataFrameGroupBy.skew: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.DataFrameGroupBy.take: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.DataFrameGroupBy.take: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.DataFrameGroupBy.tshift: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.DataFrameGroupBy.tshift: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.SeriesGroupBy.nlargest: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.SeriesGroupBy.nlargest: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.SeriesGroupBy.nsmallest: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.SeriesGroupBy.nsmallest: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.SeriesGroupBy.nunique: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.SeriesGroupBy.nunique: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.SeriesGroupBy.unique: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.SeriesGroupBy.unique: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.SeriesGroupBy.value_counts: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.SeriesGroupBy.value_counts: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.SeriesGroupBy.is_monotonic_increasing: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.SeriesGroupBy.is_monotonic_increasing: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.SeriesGroupBy.is_monotonic_decreasing: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.SeriesGroupBy.is_monotonic_decreasing: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.groupby.DataFrameGroupBy.corrwith: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.groupby.DataFrameGroupBy.corrwith: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.resample.Resampler.pipe: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.resample.Resampler.nunique: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.resample.Resampler.nunique: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
pandas.core.resample.Resampler.first: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.resample.Resampler.last: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.resample.Resampler.max: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.resample.Resampler.min: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.resample.Resampler.prod: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
pandas.core.resample.Resampler.sum: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
```

**Comment by giantpanpan** [comment]
Hi @datapythonista Can I work on this issue?

**Comment by datapythonista** [comment]
Absolutely, let me know if you need help. Thanks @giantpanpan

**Comment by giantpanpan** [comment]
@datapythonista  Thank you!

**Comment by datapythonista** [comment]
We updated that document, but you're checking the documentation of the last stable release (which makes sense, but in this case is outdated).

Once you have conda set up, from the pandas project root, you need to simply run `conda env create`. And after it's created activate the environment with `source activate pandas-dev` (not sure if in windows is `conda activate pandas-dev`).

**Comment by giantpanpan** [comment]
@datapythonista I see! Thanks!

**Comment by datapythonista** [comment]
this is the latest version of the contributing page: https://pandas-docs.github.io/pandas-docs-travis/contributing.html

**Comment by aqurilla** [comment]
Hi @datapythonista , I am a first-time contributor and was wondering if I could also work on this issue? 
I see that someone else is already working on this - what would be a good way to ensure no overlap?
Thanks!

**Comment by LJArendse** [comment]
@datapythonista is it okay if I pick up this issue?

**Comment by LJArendse** [comment]
@datapythonista awesome, will do, thanks

**Comment by datapythonista** [comment]
@giantpanpan are you still working on this?

**Comment by iam-mhaseeb** [comment]
@datapythonista  Anyone working on it? If no one is working I can work on it.

**Comment by datapythonista** [comment]
go ahead, thank you!

**Comment by iam-mhaseeb** [comment]
Okay. I'm working on it.

**Comment by clham** [comment]
It looks like there are bunch of these left. I'm having some difficulty figuring this out: in the case where the doc ends with something touched by `@substitution` that ends in a new line, the new-lines in the main doc, and the substitution both "stay", and cause the raw_doc to end in a double new-line (causing the validation to fail). An example is https://github.com/pandas-dev/pandas/blob/3855a27be4f04d15e7ba7aee12f0220c93148d3d/pandas/core/generic.py#L4965
What's the desired outcome on this edge?

**Comment by schimizzimj** [comment]
This hasn't been active for a couple of months, so what's the status with this issue? Is there still work to be done?

**Comment by alexcwatt** [comment]
@schimizzimj I think a couple docstrings were overlooked but I've just opened a PR which should close this out.

**Comment by c0dr7** [comment]
I am noobie and this looks like an easy tast to start but I would like some guidance maybe there is a noobie instructor that can point in the steps so I can start working on this?

**Comment by datapythonista** [comment]
Thanks for volunteering @c0dr7 

You can take a look at #26037, and continue from there. Check the comments from the review to see what was incomplete, missing or wrong.

The general idea is to replace cases like:
```
def foo():
    """Compute count of group, excluding missing values"""
    pass
```

by:
```
def foo():
    """
    Compute count of group, excluding missing values
    """
    pass
```

But in some cases the docstrings are generated dynamically and it's trickier. Feel free to send a partial PR with the easy cases, and leave the more complex for a second PR.

**Comment by abdurrehman11** [comment]
@datapythonista I am newbie to open source contribution but not in pandas. I have successfully setup an environment locally using pandas contribution docs. I am not seeing any good first issue on which no one is working. Can you please refer me to some good first issue? Excited to hear from you soon.

**Comment by TomAugspurger** [comment]
Take a look at
https://github.com/pandas-dev/pandas/labels/good%20first%20issue. If
someone said they were going to work on it, but haven't progressed
in a couple weeks and there isn't an open pull request for it, then I think
it's fine for you to work on it.

**Comment by killerontherun1** [comment]
Hey, 

Newbie, again. I setup the environment and I think I have the latest version.  

I ran: 
`./scripts/validate_docstrings.py --prefix=pandas.core --errors=GL01,GL02` 

Output: 
`pandas.core.resample.Resampler.pipe: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)`
`pandas.core.groupby.GroupBy.pipe: Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)`
`pandas.core.groupby.GroupBy.pipe: Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)`

I'm going to go ahead and correct these and submit a PR for the same. 

PS: I'm looking for someone to correct me if I'm wrong in submitting my first PR. Thanks.
